render_block/django.py

```diff
@@ -30,24 +33,38 @@ def django_render_block(template, block_name, context, request=None):
 
     # Get the underlying django.template.base.Template object.
     template = template.template
+    cache_key = _make_node_cache_key(template, block_name)
 
     # Bind the template to the context.
     with context_instance.bind_template(template):
-        # Before trying to render the template, we need to traverse the tree of
-        # parent templates and find all blocks in them.
-        parent_template = _build_block_context(template, context_instance)
-
         try:
-            return _render_template_block(template, block_name, context_instance)
-        except BlockNotFound:
-            # The block wasn't found in the current template.
+            node, render_context = _NODES_CACHE[cache_key]
+        except KeyError:
+            # Before trying to render the template, we need to traverse the tree of
+            # parent templates and find all blocks in them.
+            parent_template = _build_block_context(template, context_instance)
+
+            try:
+                node, render_context = _find_template_block(template, block_name, context_instance)
+            except BlockNotFound:
+                # The block wasn't found in the current template.
+
+                # If there's no parent template (i.e. no ExtendsNode), re-raise.
+                if not parent_template:
+                    raise
+
+                # Check the parent template for this block.
+                node, render_context = _find_template_block(parent_template, block_name, context_instance)
+
+            if not settings.DEBUG:
```

Comment:
I don't think this is quite right, for the [Django template engine](https://docs.djangoproject.com/en/4.0/topics/templates/#django.template.backends.django.DjangoTemplates) it can be overridden for each backend separately (i.e. you can have `DEBUG` be `False` and individually enable `debug` for the specific engine).

I think the following will do it though:

```suggestion
            if not template.engine.debug:
```

Reply:
Great! I didn't knew that..